projects/ngneat/transloco/src/lib/transloco.config.ts

```diff
@@ -32,6 +32,7 @@ export const defaultConfig: TranslocoConfig = {
   defaultLang: 'en',
   reRenderOnLangChange: false,
   prodMode: false,
+  fallbackLang: 'en',
```

Comment:
With this PR, we evaluate fallback languages eagerly, which is before loading the first choice translation file.
But previously, fallback languages are evaluated lazily after loading translation file fails.

Then, there would be a breaking change (if w/o setting `fallbackLang: 'en'`) that:

- Currently, if a user uses the `DefaultFallbackStrategy` but not set `fallbackLang`, then `DefaultFallbackStrategy` throws an exception directly that says: `When using the default fallback, a fallback language must be provided in the config!`.

  But previously, if a user uses the `DefaultFallbackStrategy` but not set `fallbackLang`. The exception is only thrown when loading translation file fails, instead of before loading the first choice translation file.

  ```ts
  export class DefaultFallbackStrategy implements TranslocoFallbackStrategy {
    constructor(
      @Inject(TRANSLOCO_CONFIG) private userConfig: TranslocoConfig
    ) {}

    getNextLangs(failedLang: string) {
      const fallbackLang = this.userConfig.fallbackLang;
      if (!fallbackLang) {
        throw new Error(
          'When using the default fallback, a fallback language must be provided in the config!'
        );
      }

      return Array.isArray(fallbackLang) ? fallbackLang : [fallbackLang];
    }
  }
  ```

- Another evidence is that the existing unit tests will have 34 failures due to no `fallbackLang` found in default config.

This issue could be mitigate temporarily by adding `fallbackLang: 'en'` in transloco's default config.

Any idea about it?

Reply:
Why aren't we calculating the fallback on fail? before calling the handler?

Reply:
@shaharkazaz Fixed.

I refreshed my mind. I agree with you, we should calculate the fallback languages when failure. So it keeps the same behavior with original one. 😄